app/transport/rest/user.go

```diff
@@ -60,3 +62,54 @@ func (uh UserHandler) CreateUser() http.Handler {
 		uh.logger.Debugw("User successfully created", "ID", id)
 	})
 }
+
+// DeleteUser will handle user creation
+func (uh UserHandler) DeleteUser() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		var user entities.UserID
+
+		if err := json.NewDecoder(req.Body).Decode(&user); err != nil {
+			WriteJSONResponse(w, http.StatusBadRequest, Response{Message: MsgBadRequest, Details: err.Error()}, uh.logger)
+			uh.logger.Errorf("Failed decoding JSON from request %+v: %+v", req, err)
+
+			return
+		}
+
+		defer func() {
+			if err := req.Body.Close(); err != nil {
+				uh.logger.Warnf("Failed closing request %+v: %+v", req, err)
+			}
+		}()
+
+		if err := user.ValidateUUID(); err != nil {
+			uh.logger.Warnf("Invalid ID in the request: %s", user.ID)
+			WriteJSONResponse(w, http.StatusBadRequest, Response{Message: MsgBadRequest, Details: err.Error()}, uh.logger)
+
+			return
+		}
+
+		if isExists, err := uh.usecase.IsUserExists(user); isExists == true {
+			err = uh.usecase.Delete(user)
```

Comment:
Is there any way how we can avoid checking if user exists before deletion?
I don't want to have 2 calls to the DB if I can make the only.

Reply:
Changed the code so that existence and deletion are performed in one call to the database